### PR TITLE
[KYUUBI #6291] Avoid NPE in MySQLErrPacket

### LINKED ISSUE #6291: [TASK][TRIVIAL] Avoid NPE in MySQLErrPacket
### What's the level of this task?

TRIVIAL

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/343) and found no similar tasks.

### Mentor

- [X] I have sufficient expertise on this task, and I volunteer to be a mentor of this task to guide contributors through the task.

### Skill requirements

+ Basic knowledge on scala

### Background and Goals

NullPointerException may occur in MySQLErrPacket

```
- transfer connection url when opening connection
java.lang.NullPointerException
	at org.apache.kyuubi.server.mysql.MySQLErrPacket$.apply(MySQLGenericPackets.scala:57)
	at org.apache.kyuubi.server.mysql.MySQLCommandHandler.exceptionCaught(MySQLCommandHandler.scala:63)
	at org.apache.kyuubi.server.mysql.MySQLCommandHandler.$anonfun$channelRead0$2(MySQLCommandHandler.scala:86)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```

### Implementation steps

Add non-null detection for `e.getMessage`

### Additional context

Introduction of [2024H1 Kyuubi Code Contribution Program](https://github.com/apache/kyuubi/issues/6232)

### PR DESCRIPTION
**Pull request by zhaohehuhu**
# :mag: Description
as title
## Issue References 🔗

This pull request fixes #6291

## Describe Your Solution 🔧

Add a rule to check if it's null 

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by zhaohehuhu** [comment]
@wForget  plz review it. Thanks.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/mysql/MySQLGenericPackets.scala [hunk 1]
```diff
@@ -54,7 +54,8 @@ object MySQLErrPacket {
       case kse: KyuubiSQLException if kse.getCause != null =>
         // prefer brief nested error message instead of whole stacktrace
         apply(kse.getCause)
-      case e: Exception if e.getMessage contains "NoSuchDatabaseException" =>
+      case e: Exception
+          if (e.getMessage != null) && (e.getMessage contains "NoSuchDatabaseException") =>
```
Comment (wForget) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/mysql/MySQLGenericPackets.scala:58]:
```suggestion
          if e.getMessage != null && e.getMessage.contains("NoSuchDatabaseException") =>
```

Comment (zhaohehuhu) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/mysql/MySQLGenericPackets.scala:58]: Thanks !

**Comment by wForget** [comment]
Please complete the checklist

**Review by cxzl25 — APPROVED** [review]

**Review by yikf — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master(1.10.0)/1.9.1/1.8.2/1.7.4

**Pull request closed without merging**